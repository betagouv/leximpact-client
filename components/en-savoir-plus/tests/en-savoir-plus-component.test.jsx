import { createShallow } from "@material-ui/core/test-utils";

import EnSavoirPlusComponent from "../en-savoir-plus-component";

describe("components | en-savoir-plus-component", () => {
  let shallow;
  beforeAll(() => {
    const options = { dive: true };
    shallow = createShallow(options);
  });

  describe("snapshot", () => {
    it("doit correspondre avec les props obligatoires", () => {
      const props = {
        onClosePopin: jest.fn(),
      };
      const wrapper = shallow(<EnSavoirPlusComponent {...props} />);
      expect(wrapper).toBeDefined();
      expect(wrapper).toMatchSnapshot();
    });
  });
});
