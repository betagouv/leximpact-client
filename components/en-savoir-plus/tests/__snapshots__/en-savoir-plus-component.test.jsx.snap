// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components | en-savoir-plus-component snapshot doit correspondre avec les props obligatoires 1`] = `
<div
  className="EnSavoiPlusComponent-container-3"
>
  <WithStyles(IconButton)
    className="EnSavoiPlusComponent-closeButton-2"
    onClick={[MockFunction]}
  >
    <pure(CloseIcon)
      fontSize="small"
    />
  </WithStyles(IconButton)>
  <WithStyles(Grid)
    alignItems="flex-start"
    container={true}
    direction="row"
    justify="space-between"
  >
    <WithStyles(Grid)
      className="EnSavoiPlusComponent-gridItemText-6"
      item={true}
      xs={6}
    >
      <WithStyles(Typography)>
        <span
          className="EnSavoiPlusComponent-spanTitreIntro-8"
        >
          LexImpact, c'est quoi ?
        </span>
      </WithStyles(Typography)>
      <WithStyles(Typography)>
        LexImpact IR est une interface qui permet de
        <b>
          simuler, de façon rapide, l'impact des réformes paramétriques de l'impôt sur le revenu.
        </b>
        L'estimation des impacts sur des foyers fiscaux types est accessible à tous ; les mesures d'impacts sur la population française et les recettes de l'État sont disponibles en accès restreint.
      </WithStyles(Typography)>
      <a
        className="EnSavoiPlusComponent-alink-1"
        href="/presentation-et-cgu"
      >
        en savoir plus
      </a>
    </WithStyles(Grid)>
    <WithStyles(Grid)
      className="EnSavoiPlusComponent-gridItemButtons-5"
      item={true}
      xs={6}
    >
      <div>
        <div
          className="EnSavoiPlusComponent-divButton-4"
        >
          <Link
            href="/presentation-et-cgu"
          >
            <WithStyles(Button)
              color="inherit"
              fullWidth={true}
              size="large"
              variant="outlined"
            >
              Présentation et conditions d'utilisation
            </WithStyles(Button)>
          </Link>
        </div>
        <div
          className="EnSavoiPlusComponent-divButton-4"
        >
          <Link
            href="/mentions-legales"
          >
            <WithStyles(Button)
              color="inherit"
              fullWidth={true}
              size="large"
              variant="outlined"
            >
              Mentions légales
            </WithStyles(Button)>
          </Link>
        </div>
        <div
          className="EnSavoiPlusComponent-divButton-4"
        >
          <WithStyles(Button)
            color="inherit"
            fullWidth={true}
            href="mailto:leximpact@an.fr"
            size="large"
            variant="outlined"
          >
            Une question ? Un bug ? Un avis ?
          </WithStyles(Button)>
          <p
            className="EnSavoiPlusComponent-pAdresseMail-7"
          >
            Écrivez-nous à leximpact@an.fr !
          </p>
        </div>
      </div>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</div>
`;
